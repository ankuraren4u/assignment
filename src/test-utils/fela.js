import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createRenderer } from "fela";
import { RendererProvider, ThemeProvider } from "react-fela";

import { theme } from "./../theme";
Enzyme.configure({ adapter: new Adapter() });

const renderer = createRenderer();

export const felaShallow = node =>
  shallow(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );

export const felaMount = node =>
  mount(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );
