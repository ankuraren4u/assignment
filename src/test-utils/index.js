export { felaShallow, felaMount } from "./fela";
