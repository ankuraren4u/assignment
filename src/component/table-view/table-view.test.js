import { TableView } from ".";
import {felaShallow, felaMount} from "../../test-utils/";

const obj = {
    testKey1: "testValue1",
    testKey2: "testValue2"
}

describe('<TableView />', () => {

  describe('By default', () => {
    let component;
    beforeAll(() => {
      component = felaShallow(<TableView obj={obj} />);
    });
    
    it('renders', () => {
      expect(component.html()).toMatchSnapshot();
    }, );
  });
})