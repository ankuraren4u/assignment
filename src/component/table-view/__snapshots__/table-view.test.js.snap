// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableView /> By default renders 1`] = `"<div class=\\"table-view a b c d e\\"><div class=\\"f a g h i j k\\"><div class=\\"b\\">testKey1</div><div class=\\"l\\">testValue1</div></div><div class=\\"f a g h i j k\\"><div class=\\"b\\">testKey2</div><div class=\\"l\\">testValue2</div></div></div>"`;
